vms/platformvm/txs/fee/calculator.go

```diff
@@ -13,24 +13,26 @@ import (
 
 var _ txs.Visitor = (*calculator)(nil)
 
-func NewStaticCalculator(config StaticConfig, upgradeTimes upgrade.Config) Calculator {
-	return Calculator{
+func NewStaticCalculator(config StaticConfig, upgradeTimes upgrade.Config, chainTime time.Time) *Calculator {
```

Comment:
Pass to the constructor all quantities that won't change.for different transactions in a single block and only those.
Pass to CalculateFee all quantities that will change for different transactions in a single block, and only those.